### ISSUE TITLE
`name` is actually not reserved keyword in Postgres based on official Postgres document

### ISSUE THREAD

**Issue by hongbo-miao** [sqlfluff/sqlfluff#7269]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

(Following up https://github.com/sqlfluff/sqlfluff/issues/7233 it was closed without a clear explanation.)


Based on official Postgres document at https://www.postgresql.org/docs/current/sql-keywords-appendix.html, `name` is not reserved keyword.

<img width="2005" height="266" alt="Image" src="https://github.com/user-attachments/assets/db0fda35-369e-4e43-acc2-144278ad194a" />


When upgrade sqlfluff from `3.4.2` to `3.5.0`, I got errors such as

```sh
== [up.sql] FAIL                                                                                                                                                                                                      
L:  13 | P:   5 | RF04 | Keywords should not be used as identifiers.                                                                                                                                                                                                                                     
                       | [references.keywords]
```

for line


```sql
create or replace function trigger_update_updated_at()
returns trigger as
$$
begin
  new.updated_at = now();
  return new;
end;
$$ language plpgsql;

create table if not exists opal_client
(
    id uuid default gen_random_uuid(),
    name text not null,  -- This line caused the issue
    config text,
    created_at timestamptz not null default now(),
    updated_at timestamptz not null default now(),
    primary key (id)
);
```

### Expected Behaviour

Expect no errors.

### Observed Behaviour

Throw errors.


### How to reproduce


Run

```sh
sqlfluff lint --dialect=postgres folder
```

with file `up.sql` inside.

```sql
create or replace function trigger_update_updated_at()
returns trigger as
$$
begin
  new.updated_at = now();
  return new;
end;
$$ language plpgsql;

create table if not exists opal_client
(
    id uuid default gen_random_uuid(),
    name text not null,  -- This line caused the issue
    config text,
    created_at timestamptz not null default now(),
    updated_at timestamptz not null default now(),
    primary key (id)
);
```


### Dialect

postgres

### Version


```sh
sqlfluff, version 3.5.0
```

### Configuration


.sqlfluff
```
[sqlfluff]
exclude_rules = layout.long_lines
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by WittierDinosaur** [comment]
RF04 is for **non-reserved keywords**. Reserved keywords don't parse. Non-reserved keywords parse but raise errors. It sounds like you just don't want the RF04 rule, in which case, turn it off in your config. name is non-reserved in sqlfluff

**Comment by hongbo-miao** [comment]
Ah, I see now. Even though `name` is not a reserved word, it is still considered a keyword and using it is not recommended according to RF04: https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-references.keywords

> Keywords should not be used as identifiers.
> Although unreserved keywords can be used as identifiers, and reserved words can be used as quoted identifiers, best practice is to avoid where possible, to avoid any misunderstandings as to what the alias represents.

Thank you for explaining!
